README.md

```diff
@@ -756,12 +756,201 @@ Each of the `resource-limits` settings below contain two numeric fields: `hard-l
+</tr>
+<tr>
+  <td><code>settings.oci-defaults.resource-limits.max-open-files.hard-limit</code></td>
+  <td>1048576</td>
+</tr>
+<tr>
+  <td rowspan="2"><code>RLIMIT_SIGPENDING</code></td>
+  <td><code>settings.oci-defaults.resource-limits.max-pending-signals.soft-limit</code></td>
+  <td>-</td>
+  <td rowspan="2">signals</td>
+</tr>
+<tr>
+  <td><code>settings.oci-defaults.resource-limits.max-pending-signals.hard-limit</code></td>
+  <td>-</td>
+</tr>
+<tr>
+  <td rowspan="2"><code>RLIMIT_NPROC</code></td>
+  <td><code>settings.oci-defaults.resource-limits.max-processes.soft-limit</code></td>
+  <td>-</td>
+  <td rowspan="2">processes</td>
+</tr>
+<tr>
+  <td><code>settings.oci-defaults.resource-limits.max-processes.hard-limit</code></td>
+  <td>-</td>
+</tr>
+<tr>
+  <td rowspan="2"><code>RLIMIT_RTPRIO</code></td>
+  <td><code>settings.oci-defaults.resource-limits.max-realtime-priority.soft-limit</code></td>
+  <td>-</td>
+  <td rowspan="2">-</td>
+</tr>
+<tr>
+  <td><code>settings.oci-defaults.resource-limits.max-realtime-priority.hard-limit</code></td>
+  <td>-</td>
+</tr>
+<tr>
+  <td rowspan="2"><code>RLIMIT_RTTIME</code></td>
+  <td><code>settings.oci-defaults.resource-limits.max-realtime-timeout.soft-limit</code></td>
+  <td>-</td>
+  <td rowspan="2">microseconds</td>
+</tr>
+<tr>
+  <td><code>settings.oci-defaults.resource-limits.max-realtime-timeout.hard-limit</code></td>
+  <td>-</td>
+</tr>
+<tr>
+  <td rowspan="2"><code>RLIMIT_RSS</code></td>
+  <td><code>settings.oci-defaults.resource-limits.max-resident-set.soft-limit</code></td>
+  <td>-</td>
+  <td rowspan="2">bytes</td>
+</tr>
+<tr>
+  <td><code>settings.oci-defaults.resource-limits.max-resident-set.hard-limit</code></td>
+  <td>-</td>
+</tr>
+<tr>
+  <td rowspan="2"><code>RLIMIT_STACK</code></td>
+  <td><code>settings.oci-defaults.resource-limits.max-stack-size.soft-limit</code></td>
+  <td>-</td>
+  <td rowspan="2">bytes</td>
+</tr>
+<tr>
+  <td><code>settings.oci-defaults.resource-limits.max-stack-size.hard-limit</code></td>
+  <td>-</td>
+</tr>
+</table>
+
+Any integer between 0 to 9223372036854775807 (2 ^63^ -1), -1 and `unlimited` are valid values to set hard-limit and soft limit for a resource. 
+The string `unlimited` or -1 as resource-limit denotes limit as unlimited. Following are the examples to set hard-limit and soft-limit:
+* Setting any valid integer value as limit:
+  ```toml
+  [settings.oci-defaults.resource-limits.<rlimit>>]
+  hard-limit = 65536
+  soft-limit = 65536
+  ```
+* Setting max possible integer value as limit:
+  ```toml
+  [settings.oci-defaults.resource-limits.<rlimit>>]
+  hard-limit = 9223372036854775807
+  soft-limit = 65536
+  ```
+* Setting unlimited value as limit:
+  ```toml
+  [settings.oci-defaults.resource-limits.<rlimit>>]
+  hard-limit = "unlimited"
+  soft-limit = 65536
+  ```
+* Setting unlimited value as limit:
+  ```toml
+  [settings.oci-defaults.resource-limits.<rlimit>>]
+  hard-limit = -1
+  soft-limit = 65536
+  ```
```

Comment:
We can support this in code, but don't need to document it. The "unlimited" one is more obvious and should be preferred.